tests/robustness/client/client.go

```diff
@@ -371,3 +372,49 @@ func toWatchEvent(event clientv3.Event) (watch model.WatchEvent) {
 	}
 	return watch
 }
+
+type ClientSet struct {
+	idProvider identity.Provider
+	baseTime   time.Time
+
+	open    bool
+	clients []*RecordingClient
+	reports []report.ClientReport
+}
+
+func NewSet(ids identity.Provider, baseTime time.Time) *ClientSet {
+	return &ClientSet{
+		idProvider: ids,
+		baseTime:   baseTime,
+
+		open:    true,
+		clients: []*RecordingClient{},
+	}
+}
+
+func (cs *ClientSet) NewClient(endpoints []string) (*RecordingClient, error) {
+	if !cs.open {
+		return nil, errors.New("The ClientSet is already closed")
+	}
+	cli, err := NewRecordingClient(endpoints, cs.idProvider, cs.baseTime)
+	if err != nil {
+		return nil, err
+	}
+	cs.clients = append(cs.clients, cli)
+	return cli, nil
+}
+
+func (cs *ClientSet) Reports() []report.ClientReport {
+	cs.reports = make([]report.ClientReport, 0, len(cs.clients))
```

Comment:
Why this sets reports but not uses it?

Reply:
This currently collects the reports from all the clients and returns it - kinda following what's proposed on the issue.  The returned client reports are going to be combined with the watch and database empty check reports in `runTraffic`.  Do you expect the reports to be used differently than just being returned?  If so, let me know and I can add in.

Reply:
Ok, I made `Reports()` close the client set if it's not closed and return the reports.  The reports are collected when `Close()` is called.